### PR TITLE
kubetcl create --validate crashes when no apiVersion or kind is provided

### PR DESCRIPTION
**Pull request by feihujiang**
#9224

### CODE REVIEW

File: pkg/kubectl/resource/visitor.go [hunk 1]
```diff
@@ -276,7 +276,7 @@ func (v *DirectoryVisitor) Visit(fn VisitorFunc) error {
 			return fmt.Errorf("unable to read %q: %v", path, err)
 		}
 		if err := ValidateSchema(data, v.Schema); err != nil {
-			return err
+			return fmt.Errorf("error validating %q: %v", Path, err)
```
Comment (krousey) [inline pkg/kubectl/resource/visitor.go]: _output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource/visitor.go:279: undefined: Path

**Comment by jlowdermilk** [comment]
@mikedanese, this appears to overlap with fixes you've been working on.

**Comment by feihujiang** [comment]
@mikedanese @jlowdermilk @krousey  I changed something, could you please review this PR again?

**Comment by mikedanese** [comment]
@feihujiang sure I'll try to get to this tonight. I know there are no conflicts, but could you rebase onto master?

File: pkg/kubectl/resource/visitor.go [hunk 2]
```diff
@@ -276,7 +276,7 @@ func (v *DirectoryVisitor) Visit(fn VisitorFunc) error {
 			return fmt.Errorf("unable to read %q: %v", path, err)
 		}
 		if err := ValidateSchema(data, v.Schema); err != nil {
-			return err
+			return fmt.Errorf("error validating %q: %v", path, err)
```
Comment (mikedanese) [inline pkg/kubectl/resource/visitor.go]: This should only return an error if !v.IgnoreErrors (see below)

File: pkg/api/validation/schema.go [hunk 3]
```diff
@@ -77,9 +77,15 @@ func (s *SwaggerSchema) ValidateBytes(data []byte) error {
...
-	kind := fields["kind"].(string)
-	return s.ValidateObject(obj, apiVersion, "", apiVersion+"."+kind)
+	apiVersion := fields["apiVersion"]
+	if apiVersion == nil {
+		fmt.Errorf("apiVersion not set")
+	}
+	kind := fields["kind"]
+	if kind == nil {
+		fmt.Errorf("kind not set")
+	}
```
Comment (mikedanese) [inline pkg/api/validation/schema.go]: why not use  runtime.UnstructuredJSONScheme.DataVersionAndKind for this?

**Comment by feihujiang** [comment]
@mikedanese  thanks. But I rebased onto master just ten minutes ago, and it shows this branch is even with GoogleCloudPlatform:master. so I don't think it need to rebase onto master.

**Comment by mikedanese** [comment]
@feihujiang oops. It looked like some of the logging was out of date. My mistake.

**Comment by mikedanese** [comment]
This looks good. @k8s-bot ok to test

File: pkg/api/validation/schema.go (hunk 3 above)
Comment (feihujiang) [inline pkg/api/validation/schema.go]: err := json.Unmarshal(data, &obj), this already parses json package, I think we can't add an new unmarshal call, so I don't use runtime.UnstructuredJSONScheme.DataVersionAndKind. what's your point?

Comment (mikedanese) [inline pkg/api/validation/schema.go]: DataVersionAndKind does exactly this. It would be cleaner to use it but less efficient. Just entertaining the idea.

**Pull request merged**